### PR TITLE
fix(dashboard): handle PTY replay disconnect

### LINKED ISSUE #63394: Dashboard PTY reattach logs uncaught WebSocketDisconnect during snapshot replay
## Bug

The keep-alive PTY route replays buffered output before entering its existing WebSocket disconnect/cleanup block. If the client disconnects during that replay, `send_bytes()` raises `WebSocketDisconnect`, Uvicorn logs an ASGI exception, and the route skips its normal registry detach.

Current `origin/main` (`7b5ba20`):

```python
await session.attach(ws)

try:
    while True:
        msg = await ws.receive()
        ...
except WebSocketDisconnect:
    pass
finally:
    PTY_REGISTRY.detach(attach_token, ws)
```

`PtySession.attach()` performs the vulnerable replay:

```python
snap = self.buffer.snapshot()
if snap:
    await ws.send_bytes(snap)
```

Observed exception path:

```text
pty_ws -> session.attach -> ws.send_bytes(snap)
starlette.websockets.WebSocketDisconnect: 1006
```

## Scope correction

This exception is **nonfatal to Uvicorn** and does not terminate the dashboard service. A service outage initially associated with this traceback was caused by a separate explicit process termination. This issue only tracks the uncaught per-connection exception and missed cleanup.

## Expected behavior

A client disappearing during initial replay should be treated like any other WebSocket disconnect: no ASGI error log, and the registry should detach the socket.

## Minimal fix

Move `await session.attach(ws)` inside the route's existing `try / except WebSocketDisconnect / finally` block. This catches only the expected transport exception and preserves cleanup without adding a broad catch to `PtySession`.

A deterministic regression test can make `session.attach()` raise `WebSocketDisconnect` and assert that `PTY_REGISTRY.detach()` is called.

### PR DESCRIPTION
**Pull request by kaykayyali**
## What does this PR do?

Handles a client disconnect during keep-alive PTY snapshot replay at the WebSocket route boundary that already owns disconnect handling and registry cleanup.

`session.attach(ws)` currently runs immediately before `pty_ws` enters its existing `try / except WebSocketDisconnect / finally detach` block. Moving that call inside the block prevents the expected disconnect from escaping as an ASGI error and guarantees the registry detaches the socket.

This keeps `PtySession.attach()` transport-agnostic and avoids a broad exception catch or duplicate state rollback.

## Related Issue

Fixes #63394

Related to #63193, which addresses the same race with additional session-layer rollback. This is an intentionally narrower alternative using the route's existing cleanup boundary.

## Type of Change

- [x] 🐛 Bug fix (non-breaking change that fixes an issue)
- [ ] ✨ New feature (non-breaking change that adds functionality)
- [ ] 🔒 Security fix
- [ ] 📝 Documentation update
- [x] ✅ Tests (adding or improving test coverage)
- [ ] ♻️ Refactor (no behavior change)
- [ ] 🎯 New skill (bundled or hub)

## Changes Made

- `hermes_cli/web_server.py`: include initial PTY snapshot replay in the existing WebSocket disconnect/cleanup boundary.
- `tests/test_pty_keepalive_ws.py`: add a deterministic regression proving disconnect during attach returns cleanly and calls `PTY_REGISTRY.detach()`.

## How to Test

1. Run the focused regression:
   ```bash
   venv/bin/python -m pytest tests/test_pty_keepalive_ws.py::test_disconnect_during_snapshot_detaches_session -q
   ```
2. Run the relevant PTY/dashboard suites:
   ```bash
   scripts/run_tests.sh tests/test_pty_session.py tests/test_pty_keepalive_ws.py tests/test_web_server.py
   ```
3. Run the full suite:
   ```bash
   scripts/run_tests.sh
   ```

Expected: the regression and existing suites pass; a `WebSocketDisconnect` raised by `session.attach()` does not escape, and registry detach is called for the attach token.

### Code

- [x] I've read the [Contributing Guide](https://github.com/NousResearch/hermes-agent/blob/main/CONTRIBUTING.md)
- [x] My commit messages follow [Conventional Commits](https://www.conventionalcommits.org/) (`fix(scope):`, `feat(scope):`, etc.)
- [x] I searched for [existing PRs](https://github.com/NousResearch/hermes-agent/pulls) and documented the overlap with #63193
- [x] My PR contains **only** changes related to this fix/feature (no unrelated commits)
- [ ] I've run `pytest tests/ -q` and all tests pass
- [x] I've added tests for my changes (required for bug fixes, strongly encouraged for features)
- [x] I've tested on my platform: Linux 7.0.0-3-pve (Proxmox LXC)

### Documentation & Housekeeping

- [x] Relevant documentation updated — N/A; no public API or user workflow change
- [x] `cli-config.yaml.example` updated — N/A; no config changes
- [x] `CONTRIBUTING.md` or `AGENTS.md` updated — N/A; no architecture or workflow changes
- [x] Cross-platform impact considered — route-level async exception handling is platform-independent
- [x] Tool descriptions/schemas updated — N/A; no tool behavior changes

## Validation

| Check | Result |
|---|---|
| Regression test before fix | Failed with uncaught `WebSocketDisconnect` |
| Regression test after fix | Passed |
| Relevant PTY/dashboard suites | 16 passed |
| Independent code review | Passed; no security, logic, or race-condition findings |
| Full suite | Completed with 6 unrelated failures: 5 reproduce on clean `origin/main`; the Mistral lazy-install failure passed on focused rerun |
| `git diff --check` | Passed |

### CODE REVIEW

**Comment by alt-glitch** [comment]
> *This was generated by AI during triage.*

Related: #63394 (the issue this fixes) and competing OPEN PR #63193, which fixes the same disconnect race at the session layer (`hermes_cli/pty_session.py` guards the ring-buffer replay) rather than at the route boundary. Same goal, different mechanism — this PR is the intentionally narrower route-cleanup-boundary variant, #63193 adds session-layer rollback. Not a duplicate; a maintainer should pick one approach.

**Comment by teknium1** [comment]
Thanks for the focused route-boundary fix. The current-main premise is valid: `PtySession.attach()` can raise during buffered replay at `hermes_cli/pty_session.py:91-93`, while `pty_ws()` currently calls it before the existing `WebSocketDisconnect`/`finally` boundary at `hermes_cli/web_server.py:15788-15825`.

The proposed move places the attach in the boundary that already catches the expected exception and always calls `PTY_REGISTRY.detach()`. The new handler-level regression directly verifies that contract. The change adds no configuration, tool, cache, or public API surface.

Automated hermes-sweeper review.
